pkg/controller/elasticsearch/driver/esstate.go

```diff
@@ -33,12 +33,12 @@ type MemoizingESState struct {
 }
 
 // NewMemoizingESState returns an initialized MemoizingESState.
-func NewMemoizingESState(esClient esclient.Client) ESState {
+func NewMemoizingESState(ctx context.Context, esClient esclient.Client) ESState {
```

Comment:
Again I think the context should not be stored, but passed into the `NodesInCluster`, `ShardAllocationsEnabled`, and `Health` methods. Those would pass the context along to initialize by creating a closure.

It probably doesn't matter greatly, but it'll allow you to attribute the memoized ES requests to the correct parent span. Depends on how much you care vs. how much work is involved in updating call sites.

Reply:
I think I agree that would be a more common thing to do.

Reply:
I can change it.

Reply:
I am having second thoughts, because this is a cached client abstraction we are using here, so repeated calls with different context would not have any effect